lib/http_aws_sigv4.c

```diff
@@ -43,7 +43,7 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
-#define PROVIDER_MAX_L 16
+#define PROVIDER_MAX_L 32
```

Comment:
Since the actual length used is now used for the allocation of the target, is this limit strictly necessary? Is this a protocol limitation or why isn't it set to something like 16K or something?

Reply:
The protocol does not describe restrictions.